File: test_data/code/python/1175-prime-arrangements.py
class Solution:
    def numPrimeArrangements(self, n: int) -> int:
        MOD = 10**9 + 7
        def is_prime(k):
            if k < 2:
                return False
            for i in range(2, int(k**0.5) + 1):
                if k % i == 0:
                    return False
            return True

        prime_count = 0
        for i in range(2, n + 1):
            if is_prime(i):
                prime_count += 1

        ans = 1
        for _ in range(prime_count):
            ans = ans * _ ** 2 % MOD
        for _ in range(n - prime_count):
            ans = ans * (_ + 1) % MOD
        return ans

File: test_data/code/java/1175-prime-arrangements.java
class Solution {
    public int numPrimeArrangements(int n) {
        int MOD = 1000000007;
        int count = 0;
        boolean[] isPrime = new boolean[n + 1];
        for (int i = 2; i <= n; i++) {
            isPrime[i] = true;
        }
        for (int i = 2; i * i <= n; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= n; j += i) {
                    isPrime[j] = false;
                }
            }
        }
        for (int i = 2; i <= n; i++) {
            if (isPrime[i]) {
                count++;
            }
        }
        long res = 1;
        for (int i = 1; i <= Math.min(count, n - count); i++) {
            res = res * i % MOD;
        }
        for (int i = 1; i <= Math.abs(count - n); i++) {
            res = res * i % MOD;
        }
        return (int) res;
    }
}